### PR TITLE
CMS-72-date-visual-component

### PR DESCRIPTION
**Pull request by KeimiLL**

### CODE REVIEW

File: frontend/src/app/shared/components/date/date.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    standalone: true,
+    imports: [CommonModule, MatIconModule],
+    templateUrl: "./date.component.html",
+    styleUrls: ["./date.component.scss"],
+})
+export class DateComponent implements OnChanges {
+    @Input() date: string;
+    cssClass: string;
+
+    ngOnChanges(): void {
```
Comment (Rolero4) [inline frontend/src/app/shared/components/date/date.component.ts:16]: ngOnChanges? Czemu nie init?

File: frontend/src/app/shared/components/date/date.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@Component({
+    selector: "app-date",
+    standalone: true,
+    imports: [CommonModule, MatIconModule],
+    templateUrl: "./date.component.html",
+    styleUrls: ["./date.component.scss"],
+})
+export class DateComponent implements OnChanges {
+    @Input() date: string;
+    cssClass: string;
```
Comment (Rolero4) [inline frontend/src/app/shared/components/date/date.component.ts:14]: accessibility

File: frontend/src/app/shared/components/date/date.component.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component({
+    selector: "app-date",
+    standalone: true,
+    imports: [CommonModule, MatIconModule],
+    templateUrl: "./date.component.html",
+    styleUrls: ["./date.component.scss"],
+})
+export class DateComponent implements OnChanges {
+    @Input() date: string;
```
Comment (Rolero4) [inline frontend/src/app/shared/components/date/date.component.ts:13]: accessibility

File: frontend/src/app/shared/components/date/date.component.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    cssClass: string;
+
+    ngOnChanges(): void {
+        const dateObject = new Date(this.date);
+        const today = new Date();
+
+        if (this.areDatesEqual(dateObject, today)) {
```
Comment (Rolero4) [inline frontend/src/app/shared/components/date/date.component.ts:20]: Coś w tym stylu, żeby uniknąć niepotrzebnego chaina dzięki szybszym returnom:?
      
   this.cssClass = this.cssClassPicker(dateObject, today);
   
    private cssClassPicker(inputDate: Date, currentDate: Date): string {
        if (this.areDatesEqual(inputDate, currentDate)) {
            return "date--today";
        } else if (inputDate < currentDate) {
            return "date--past";
        }
        return "date--future";
    }

**Review by Rolero4 — APPROVED** [review]

**Review by Domin8668 — APPROVED** [review]

**Comment by Domin8668** [comment]
Zmiany od Bartka i będzie git

**Pull request merged**